src/buffer.cpp

```diff
@@ -146,7 +146,7 @@ bool CNetBuf::Put ( const CVector<uint8_t>& vecbyData, int iInSize )
         const int iNumBlocks = /* floor */ ( iInSize / iBlockSize );
 
         // copy new data in internal buffer
-        for ( int iBlock = 0; iBlock < iNumBlocks; iBlock++ )
+        for ( long iBlock = 0; iBlock < iNumBlocks; iBlock++ )
```

Comment:
```suggestion
        // iBlock is a long to avoid overflowing a mulitplication later in the code
        for ( long iBlock = 0; iBlock < iNumBlocks; iBlock++ )
```